ibm_cloud_sdk_core/base_service.py

```diff
@@ -81,8 +81,7 @@ def __init__(self, vcap_services_name, url, username=None, password=None,
             raise ValueError('The URL shouldn\'t start or end with curly brackets or quotes. '
                              'Be sure to remove any {} and \" characters surrounding your URL')
 
-        user_agent_string = 'ibm-python-sdk-core-' + __version__ + self.get_os_info()
-        self.user_agent_header = self.set_user_agent_header(user_agent_string)
+        self.set_user_agent_header(self.build_user_agent())
```

Comment:
In general, I think we want the core to add the User-Agent header to the outgoing request ONLY if it hasn't already been added by the service function.  (I can't tell from this whether we will unconditionally add the core's User-Agent header or not).

Reply:
Oh good catch!